### PR TITLE
Enable C# 8.0 nullability for the CLI, Android and iOS projects, improve help command behaviour

### PR DESCRIPTION
**Pull request by premun**
This enables the nullability and resolves all warnings connected to that in the projects.

There were a couple of changes I had to do (arguments re-work) in order to achieve this and some I did to improve working with the code:
- Moved option definition to Argument classes. This makes the code easier to work with because the option definitions are together with the properties that they fill and their validation.
- Improved the help command so that you can run `help [command] [subcommand]` which was not possible before. Now it's easier to explore the tool using calling `help xy`. It now also works properly even when you mess up the subcommand name.
- Moved the Darwin class to the iOS project and kept the CLI project strictly CLI.
- Dedicate the `ios test t=` argument to the `targets` option. It was originally assigned to `timeout` but `timeout` is optional and `targets` not. This change allows to call `ios test` with all required args using shortcuts (`-o -a -t`)

Fixes: #16

### CODE REVIEW

File: src/Microsoft.DotNet.XHarness.CLI/CommandArguments/XHarnessCommandArguments.cs [hunk 1]
```diff
@@ -23,5 +39,67 @@ protected static string RootPath(string path)
...
+            if (invalidValues != null)
+            {
+                values = values.Where(v => !invalidValues.Contains(v));
+            }
+
+            var separator = Environment.NewLine + "\t- ";
+            IEnumerable<string?> allowedValued = values.Select(t => display?.Invoke(t) ?? t.ToString());
+
+            return separator + string.Join(separator, allowedValued);
+        }
```
Comment (mandel-macaque) [inline src/Microsoft.DotNet.XHarness.CLI/CommandArguments/XHarnessCommandArguments.cs:103]: I like this class :)

File: src/Microsoft.DotNet.XHarness.CLI/CommandArguments/XHarnessCommandArguments.cs [hunk 2]
```diff
@@ -23,5 +39,67 @@ protected static string RootPath(string path)
...
+        /// <summary>
+        /// Helper method that enables parsing of enums from string.
+        /// When an invalid value is supplied, available values are printed.
+        /// </summary>
+        /// <typeparam name="TEnum">Enum type</typeparam>
+        /// <param name="argumentName">Name of the argument that is being parsed, strictly for help printing purposes</param>
+        /// <param name="value">Value of the arg to be parsed</param>
+        /// <param name="invalidValues">List of values that should not be available to set</param>
+        /// <returns>Parsed enum value</returns>
+        protected static TEnum ParseArgument<TEnum>(string argumentName, string? value, params TEnum[]? invalidValues) where TEnum : struct, IConvertible
```
Comment (mandel-macaque) [inline src/Microsoft.DotNet.XHarness.CLI/CommandArguments/XHarnessCommandArguments.cs:52]: The IConvertible is a nice trick.

**Comment by premun** [comment]
What is missing is the Tests.Runners project

I tried turning it on for the test projects and 2 of them were fine, iOS.Shared.Tests was not. But not sure we want that for tests

**Comment by MattGal** [comment]
Change is huge! Did you test any of the android side before making the PR?

**Comment by premun** [comment]
@MattGal I did test every command and I tested calling help for every suite/command as well

**Pull request merged**